File: android/src/main/kotlin/com/example/design_engine/layer2_core/dark/ESurfaceVariantDark.kt
package com.example.design_engine.layer2_core.dark
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object ESurfaceVariantDark {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyd5
            AppTheme.STANDARD -> R.color.cstd5
            AppTheme.BLUE -> R.color.cbd5
            AppTheme.GREEN -> R.color.cgd5
            AppTheme.RED -> R.color.crd5
            AppTheme.NOTHING -> R.color.cnod5
        }
    }
}

File: lib/layer2_core/dark/e_surface_variant_dark.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/dark/system.dart';
import '../../layer1_raw/dark/standard.dart';
import '../../layer1_raw/dark/blue.dart';
import '../../layer1_raw/dark/green.dart';
import '../../layer1_raw/dark/red.dart';
import '../../layer1_raw/dark/nothing.dart';

class ESurfaceVariantDark {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyd5;
      case 'standard': return cstd5;
      case 'blue': return cbd5;
      case 'green': return cgd5;
      case 'red': return crd5;
      case 'nothing': return cnod5;
      default: return cstd5;
    }
  }
}
